File: repositorioacademico.client/src/app/models/carrera.ts
import { Catalogo } from './catalogo';

export interface Carrera {
  id: number;
  descripcion: string;
  estado?: string;
  areaConocimientoId: number;
  areaConocimiento?: string;
  lineasInvestigacion: Catalogo[];
}

export interface CrearCarreraPayload {
  descripcion: string;
  areaConocimientoId: number;
  estado?: string;
}

export interface ActualizarLineasCarreraPayload {
  lineaInvestigacionIds: number[];
}

File: RepositorioAcademico.Server/Domain/Carrera.cs
namespace RepositorioAcademico.Server.Domain
{
    public class Carrera
    {
        public int Id { get; set; }

        public int FacultadId { get; set; }

        public Facultad? Facultad { get; set; }

        public int AreaConocimientoId { get; set; }

        public AreaConocimiento? AreaConocimiento { get; set; }

        public string Descripcion { get; set; } = string.Empty;

        public string? Estado { get; set; }

        public ICollection<CarreraLineaInvestigacion> CarreraLineasInvestigacion { get; set; } = [];
    }
}
